File: Algorithm/Programmers/Level1/solution(26).py
# 최대공약수와 최소공배수

def solution(n, m):
    answer = []
    nm_max = 0 # 최대공약수
    if n > m :
        n,m = m,n
    # 최대공약수 구하기
    for i in range(n, 0, -1):
        if n%i == 0 and m%i==0:
            nm_max = i
            break
    answer.append(nm_max)
    # 최소공배수 구하기
    if max == 1: # 최대공약수가 1이면 최소공배수는 두 수의 곱
        answer.append(n*m)
    else:
        answer.append(n*m/nm_max) # 최소공배수는 두 수의 곱/최대공약수
    return answer

File: Algorithm/Programmers/Level1/solution(26).java
// 최대공약수와 최소공배수

class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];
        
        if (n > m){
            int temp = n;
            n = m;
            m = temp;
        }
        
        // 최대공약수
        for (int i = n; i > 0; i--){
            if (n%i == 0 && m%i == 0){
                answer[0] = i;
                break;
            }
        }
        
        // 최소공배수 = 두 수의 곱/최대공약수
        answer[1] = n * m / answer[0];
        
        return answer;
    }
}